ledger-core/virtual/integration/constructor_test.go

```diff
@@ -804,6 +805,171 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
 	mc.Finish()
 }
 
+// -> VCallRequest [A.New]
+// -> ExecutionStart
+// -> change pulse -> secondPulse
+// -> VStateReport [A]
+// -> VDelegatedCallRequest [A]
+// -> continue execution
+// -> change pulse -> thirdPulse
+// -> NO VStateReport
+// -> VDelegatedCallRequest [A] + first token
+// -> VCallResult [A.New] + second token
+// -> VDelegatedRequestFinished [A] + second token
+func TestVirtual_CallConstructor_WithTwicePulseChange(t *testing.T) {
+	t.Log("C5208")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+		return execution.Request.CallSiteMethod
+	})
+	server.ReplaceRunner(runnerMock)
+
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	var (
+		constructorIsolation = contract.ConstructorIsolation()
+
+		classA    = gen.UniqueGlobalRef()
+		outgoing  = server.BuildRandomOutgoingWithPulse()
+		objectRef = reference.NewSelf(outgoing.GetLocal())
+
+		firstPulse = server.GetPulse().PulseNumber
+
+		firstApprover  = gen.UniqueGlobalRef()
+		secondApprover = gen.UniqueGlobalRef()
+
+		firstExpectedToken, secondExpectedToken payload.CallDelegationToken
+	)
+
+	synchronizeExecution := synchronization.NewPoint(1)
+
+	// add ExecutionMocks to runnerMock
+	{
+		objectAResult := requestresult.New([]byte("finish A.New"), outgoing)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		runnerMock.AddExecutionMock("New").AddStart(
+			func(_ execution.Context) {
+				server.IncrementPulseAndWaitIdle(ctx)
```

Comment:
can you please move pulse changes from here to the block at the end? Leave here `synchronizeExecution.Synchronize()`. At the end in a block do the following: `syncPoint.Wait()`, `IncrementPulseAndWaitIdle()`,  `IncrementPulseAndWaitIdle()`, `syncPoint.Release()`

Reply:
done